cypress/support/commands.js

```diff
@@ -27,4 +27,19 @@ Cypress.Commands.add("login", (username, password) => {
     cy.get('#username', { timeout: 5000 }).type(username);
     cy.get('#password').type(password,{log:false});
     cy.get('#submit').click();
+} )
+Cypress.Commands.add("SortFilterSelection", () => {
+    cy.contains("Sort").click({ force: true });
+
+        // Get the open dropdown menu items.
+        cy.get("ul[role='listbox'] li button").then((elements) => {
+            // Get a random number using the max number of elements.
+            const max = elements.length;
+            const min = 0;
+            const randomNumber = Math.floor(Math.random() * (max - min) + min);
```

Comment:
This just seems like a very roundabout way to just do `Math.random() + max`, since on line 38 min is defined as `0`.
Is there a particular reason it's done this way?

Reply:
indexing in js starts with 0 , so the 1st element takes 0 index, second takes1. min is defined as 0 so that if random() gives 1, it does not go out of bounds

Reply:
@deshmukhmayur

Reply:
Math.random [1] is always exclusive of 1. So you don't need that. So you can just do `Math.random * max`.
The min and max is only needed when the min is something other than `0`.

Also, on another note, it should not be using a random number to test. All the elements there should have explicit tests.

[1] : https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random

Reply:
Hi @deshmukhmayur i was using random number so that each element test probability remains same . as if we go for every selection separate test, the test time may increase quite significantly for each of the dropdowns